de.metas.business/src/main/java/de/metas/purchasing/api/impl/BPartnerProductBL.java

```diff
@@ -37,19 +40,19 @@
 public class BPartnerProductBL implements IBPartnerProductBL
 {
 	private static final String MSG_ProductSalesExclusionError = "ProductSalesExclusionError";
-	
+
 	@Override
-	public void assertNotExcludedFromSaleToCustomer(final int productId, final int bpartnerId)
+	public void assertNotExcludedFromSaleToCustomer(@NonNull final ProductId productId, @NonNull final BPartnerId partnerId)
 	{
 		final IBPartnerProductDAO bpProductDAO = Services.get(IBPartnerProductDAO.class);
 
-		bpProductDAO.getExcludedFromSaleToCustomer(productId, bpartnerId)
+		bpProductDAO.getExcludedFromSaleToCustomer(productId, partnerId)
 				.ifPresent(this::throwException);
 	}
 
 	private void throwException(final ProductExclude productExclude)
 	{
-		final I_C_BPartner partner = loadOutOfTrx(productExclude.getBpartnerId(), I_C_BPartner.class);
+		final I_C_BPartner partner = loadOutOfTrx(productExclude.getBpartnerId().getRepoId(), I_C_BPartner.class);
```

Comment:
pls use BPartnerDAO to load the BP. More, in BPartnerDAO u shall have a method to fetch the BP Name...